sql: TestSchemaChangeRetryError failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/acba091f04f3d8ecabf51009bf394951fbd3643c

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestSchemaChangeRetryError PKG=github.com/cockroachdb/cockroach/pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1137989&tab=buildLog

```
I190214 06:22:56.618030 121333 util/stop/stopper.go:547  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] intent_resolver_batcher
1      [async] closedts-rangefeed-subscriber
I190214 06:22:56.618239 121334 util/stop/stopper.go:547  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190214 06:22:56.618241 121332 util/stop/stopper.go:547  quiescing; tasks left:
2      [async] closedts-subscription
11     [async] storage.IntentResolver: cleanup txn records
1      [async] closedts-rangefeed-subscriber
W190214 06:22:56.618294 121222 storage/intentresolver/intent_resolver.go:721  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
I190214 06:22:56.618333 121332 util/stop/stopper.go:547  quiescing; tasks left:
2      [async] closedts-subscription
10     [async] storage.IntentResolver: cleanup txn records
1      [async] closedts-rangefeed-subscriber
I190214 06:22:56.618414 121332 util/stop/stopper.go:547  quiescing; tasks left:
9      [async] storage.IntentResolver: cleanup txn records
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190214 06:22:56.618459 121332 util/stop/stopper.go:547  quiescing; tasks left:
8      [async] storage.IntentResolver: cleanup txn records
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190214 06:22:56.618500 121332 util/stop/stopper.go:547  quiescing; tasks left:
7      [async] storage.IntentResolver: cleanup txn records
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190214 06:22:56.618541 121332 util/stop/stopper.go:547  quiescing; tasks left:
6      [async] storage.IntentResolver: cleanup txn records
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190214 06:22:56.618581 121332 util/stop/stopper.go:547  quiescing; tasks left:
5      [async] storage.IntentResolver: cleanup txn records
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190214 06:22:56.618642 121332 util/stop/stopper.go:547  quiescing; tasks left:
4      [async] storage.IntentResolver: cleanup txn records
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190214 06:22:56.618697 121332 util/stop/stopper.go:547  quiescing; tasks left:
3      [async] storage.IntentResolver: cleanup txn records
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190214 06:22:56.618742 121332 util/stop/stopper.go:547  quiescing; tasks left:
2      [async] storage.IntentResolver: cleanup txn records
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190214 06:22:56.618779 121332 util/stop/stopper.go:547  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] storage.IntentResolver: cleanup txn records
1      [async] closedts-rangefeed-subscriber
I190214 06:22:56.618819 121332 util/stop/stopper.go:547  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190214 06:22:56.619667 121333 util/stop/stopper.go:547  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190214 06:22:56.620704 121334 util/stop/stopper.go:547  quiescing; tasks left:
2      [async] closedts-subscription
I190214 06:22:56.620805 121334 util/stop/stopper.go:547  quiescing; tasks left:
1      [async] closedts-subscription
I190214 06:22:56.620946 121333 util/stop/stopper.go:547  quiescing; tasks left:
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190214 06:22:56.620951 121332 util/stop/stopper.go:547  quiescing; tasks left:
2      [async] closedts-subscription
I190214 06:22:56.621058 121332 util/stop/stopper.go:547  quiescing; tasks left:
1      [async] closedts-subscription
I190214 06:22:56.621140 121333 util/stop/stopper.go:547  quiescing; tasks left:
1      [async] closedts-subscription
--- FAIL: TestSchemaChangeRetryError (0.68s)
    schema_changer_test.go:3769: err = <nil>

```

--- a/pkg/sql/schema_changer_test.go
+++ b/pkg/sql/schema_changer_test.go
@@ -3766,9 +3766,16 @@ func TestSchemaChangeRetryError(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	// The timestamp of the transaction is initialized.
 	tx, err := sqlDB.Begin()
 	if err != nil {
+		t.Fatal(err)
+	}
+
+	// The timestamp of the transaction is guaranteed to be fixed after
+	// this statement.
+	if _, err := tx.Exec(`
+		CREATE TABLE t.another (k INT PRIMARY KEY, v INT, pi DECIMAL DEFAULT (DECIMAL '3.14'));
+		`); err != nil {
 		t.Fatal(err)
 	}
 
@@ -3780,12 +3787,6 @@ func TestSchemaChangeRetryError(t *testing.T) {
 		t.Fatal(err)
 	}
 	rows.Close()
-
-	if _, err := tx.Exec(`
-		CREATE TABLE t.another (k INT PRIMARY KEY, v INT, pi DECIMAL DEFAULT (DECIMAL '3.14'));
-		`); err != nil {
-		t.Fatal(err)
-	}
 
 	if _, err := tx.Exec(`
 		CREATE UNIQUE INDEX vidx ON t.test (v);

